core/controllers/library_test.py

```diff
@@ -187,6 +188,62 @@ def test_library_handler_for_created_explorations(self):
         }, response_dict['activity_list'][0])
 
 
+class LibraryGroupPageTest(test_utils.GenericTestBase):
+
+    def test_library_group_pages(self):
+        """Test access to the top rated and recently published pages."""
+        response = self.testapp.get(feconf.LIBRARY_TOP_RATED_URL)
+        self.assertEqual(response.status_int, 200)
+
+        response = self.testapp.get(feconf.LIBRARY_RECENTLY_PUBLISHED_URL)
+        self.assertEqual(response.status_int, 200)
+
+    def test_handler_for_library_group_pages(self):
+        """Test library handler for top rated and recently published pages."""
+        response_dict = self.get_json(
+            feconf.LIBRARY_GROUP_DATA_URL,
+            {'group_name': feconf.LIBRARY_GROUP_RECENTLY_PUBLISHED})
+        self.assertDictContainsSubset({
+            'is_admin': False,
+            'is_moderator': False,
+            'is_super_admin': False,
+            'activity_list': [],
+            'preferred_language_codes': ['en'],
+            'profile_picture_data_url': None,
+        }, response_dict)
+
+        # Load a public demo exploration.
+        exp_services.load_demo('0')
+
+        response_dict = self.get_json(
+            feconf.LIBRARY_GROUP_DATA_URL,
+            {'group_name': feconf.LIBRARY_GROUP_RECENTLY_PUBLISHED})
+        self.assertDictContainsSubset({
+            'id': '0',
+            'category': 'Welcome',
+            'title': 'Welcome to Oppia!',
+            'language_code': 'en',
+            'objective': 'become familiar with Oppia\'s capabilities',
+            'status': rights_manager.ACTIVITY_STATUS_PUBLIC,
+        }, response_dict['activity_list'][0])
+
+        # Assign rating to exploration to test handler for top rated
```

Comment:
Split this into another test. Load the demo exploration and test that the response has 0 explorations. Then give it a rating and test that the response_dict has one exploration.

Then load another exploration, give it a higher rating, and test that the response_dict has two explorations and that they come out in the right order.